app/src/main/java/org/fossasia/openevent/app/core/orders/detail/SwipeController.java

```diff
@@ -0,0 +1,97 @@
+package org.fossasia.openevent.app.core.orders.detail;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.graphics.RectF;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.helper.ItemTouchHelper;
+import android.view.View;
+
+import org.fossasia.openevent.app.R;
+
+public class SwipeController extends ItemTouchHelper.SimpleCallback {
+
+    private final OrderAttendeesAdapter orderAttendeesAdapter;
+    private final OrderDetailViewModel orderDetailViewModel;
+    private final Context context;
+
+    public SwipeController(OrderDetailViewModel orderDetailViewModel, OrderAttendeesAdapter orderAttendeesAdapter, Context context) {
+        super(0, ItemTouchHelper.LEFT | ItemTouchHelper.RIGHT);
+        this.orderDetailViewModel = orderDetailViewModel;
+        this.orderAttendeesAdapter = orderAttendeesAdapter;
+        this.context = context;
+    }
+
+    @Override
+    public boolean onMove(RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder, RecyclerView.ViewHolder target) {
+        return false;
+    }
+
+    @Override
+    public int getMovementFlags(RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder) {
+        int dragFlags = 0;
+
+        if (orderDetailViewModel.getCheckedInStatus(viewHolder.getAdapterPosition()) == null)
+            makeMovementFlags(dragFlags, ItemTouchHelper.LEFT | ItemTouchHelper.RIGHT);
+
+        if (orderDetailViewModel.getCheckedInStatus(viewHolder.getAdapterPosition())) {
+            return makeMovementFlags(dragFlags, ItemTouchHelper.LEFT);
+        } else {
+            return makeMovementFlags(dragFlags, ItemTouchHelper.RIGHT);
+        }
+    }
+
+    @Override
+    public void onSwiped(RecyclerView.ViewHolder viewHolder, int direction) {
+        int position = viewHolder.getAdapterPosition();
+
+        orderDetailViewModel.toggleCheckIn(position);
+        orderAttendeesAdapter.notifyItemChanged(position);
+    }
+
+    @Override
+    public void onChildDraw(Canvas canvas, RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder,
+                            float dX, float dY, int actionState, boolean isCurrentlyActive) {
+
+        if (actionState == ItemTouchHelper.ACTION_STATE_SWIPE) {
+            Paint paintGreen = new Paint();
+            Paint paintRed = new Paint();
```

Comment:
This is being created on each draw, meaning on every frame. Goal was to initialize them only once, use as less commands as possible in on*Draw functions

Reply:
Initialize them in the constructor and you won't need to hold context as a field